tsl/src/planner.c

```diff
@@ -146,3 +149,82 @@ tsl_set_rel_pathlist(PlannerInfo *root, RelOptInfo *rel, Index rti, RangeTblEntr
 
 	ts_cache_release(hcache);
 }
+
+/*
+ * Check session variable to disable distributed insert with COPY.
+ *
+ * Allows testing and comparing different insert plans. Not a full GUC, since
+ * in most cases this is not something that users are expected to disable.
+ */
+static bool
+copy_mode_enabled(void)
+{
+	const char *enable_copy =
+		GetConfigOption("timescaledb.enable_distributed_insert_with_copy", true, false);
+
+	/* Default to enabled */
+	if (NULL == enable_copy)
+		return true;
+
+	return strcmp(enable_copy, "true") == 0;
```

Comment:
This is not meant for user consumption and won't be documented so it needs not care about `1` or `yes`, as explained in the comment. It is used _only_ for testing (in our tests). It can be elevated to a GUC later, but I don't think we should expose GUCs if we don't need to since they are public interfaces that we need to document and maintain.

Reply:
If this is for testing only then this should probably be only active in debug builds. Your code still exposes a GUC here even if it's not registered as one. Also this was brought up by multiple reviewers now.